File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, nums: List[int], k: int) -> int:
        minsum = sum(nums)
        window = len(nums) - k
        r = -1
        s = 0
        for l in range(len(nums)):
            s += nums[l]
            while l-r > window:
                r += 1
                s -= nums[r]
            if l-r == window:
                minsum = min(minsum,s)
        
        return sum(nums) - minsum
        
        

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int length = cardPoints.size() - k;
        int sum = 0;
        int start = 0;
        int ans = INT_MAX;
        
        for(int i = 0; i < cardPoints.size(); i++){
            sum += cardPoints[i];
            
            if(i - start + 1 > length){
                sum -= cardPoints[start];
                start++;
            }
            
            if(i - start + 1 == length){
                ans = min(ans,sum);
            }
        }
        
        int res = 0;
        for(int x : cardPoints){
            res += x;
        }
        
        if(ans != INT_MAX)  return res - ans;
        else return res;
        
    }
};